pkg/scheduler/framework/plugins/dynamicresources/claims.go

```diff
@@ -0,0 +1,150 @@
+func (cs *claimStore) len() int {
+	return len(cs.claims)
+}
+
+// all returns an iterator for all claims, whether they are owned by the user
+// or the scheduler. If there is a special ResourceClaim for extended resoures,
+// then it comes last.
+func (cs *claimStore) all() iter.Seq2[int, *resourceapi.ResourceClaim] {
+	return func(yield func(int, *resourceapi.ResourceClaim) bool) {
+		for i, claim := range cs.claims {
+			if !yield(i, claim) {
+				return
+			}
+		}
+	}
+}
+
+// allUserClaims returns an iterator which excludes the special ResourceClaim for extended resoures.
+func (cs *claimStore) allUserClaims() iter.Seq2[int, *resourceapi.ResourceClaim] {
+	return func(yield func(int, *resourceapi.ResourceClaim) bool) {
+		for i, claim := range cs.claims[0:cs.numUserOwned] {
+			if !yield(i, claim) {
+				return
+			}
+		}
+	}
+}
+
+// toAllocate returns an iterator for all claims which have no allocation result.
+// the index returned is the original index in the underlying claim store, it
+// may not be sequentially numbered (e.g. 0, 1, 2 ...).
+func (cs *claimStore) toAllocate() iter.Seq2[int, *resourceapi.ResourceClaim] {
+	return func(yield func(int, *resourceapi.ResourceClaim) bool) {
+		for i, claim := range cs.claims {
+			if claim.Status.Allocation != nil {
+				continue
+			}
+			if !yield(i, claim) {
+				return
+			}
+		}
+	}
+}
+
+// extendedResourceClaim returns the special ResourceClaim if there is one, otherwise nil.
+// The ResourceClaim is read-only and must be cloned before modifying it.
+func (cs *claimStore) extendedResourceClaim() *resourceapi.ResourceClaim {
+	if cs.numUserOwned == len(cs.claims) {
+		return nil
+	}
+	return cs.claims[cs.numUserOwned]
+}
+
+// noUserClaim returns true when there is no user claim.
+func (cs *claimStore) noUserClaim() bool {
+	return cs.numUserOwned == 0
+}
+
+// get returns the claim at the input index
+func (cs claimStore) get(i int) *resourceapi.ResourceClaim {
+	return cs.claims[i]
+}
+
+// set sets the input claim at the input index for the internal claims slice.
+func (cs claimStore) set(i int, c *resourceapi.ResourceClaim) {
+	cs.claims[i] = c
+}
+
+// updateExtendedResourceClaim updates the input claim as extended resource
+// claim in the internal claims slice.
+// It returns error when there is no extended resource claim in the internal
+// claims slice.
+func (cs *claimStore) updateExtendedResourceClaim(c *resourceapi.ResourceClaim) error {
+	if cs.numUserOwned == len(cs.claims) {
+		return fmt.Errorf("no extended resource claim")
+	}
+	if cs.claims[cs.numUserOwned].UID != c.UID {
+		cs.extendedResourceClaimTmpUID = cs.claims[cs.numUserOwned].UID
+	}
+	cs.claims[cs.numUserOwned] = c
+	return nil
+}
+
+// extendedResourceClaimUID returns the in-memory, temporary extended resource claim UID.
+// It can only be called when extended resource claim exists.
+func (cs claimStore) extendedResourceClaimUID() types.UID {
+	if cs.extendedResourceClaimTmpUID == types.UID("") {
```

Comment:
I bet you could compare against `""`, without the type cast. Constants like "" or 0 are more flexible about their type than variables.

But see my other comment about the semantic and name of this method.